src/cpp/session/SessionConsoleProcess.cpp

```diff
@@ -94,7 +94,21 @@ core::system::ProcessOptions ConsoleProcess::createTerminalProcOptions(
 #else
    core::system::setHomeToUserProfile(&shellEnv);
 #endif
-
+   
+   // don't forward GIT_ASKPASS or SSH_ASKPASS if it's set to our rpostback
+   // handlers -- those aren't required in a terminal, as the user can be
+   // prompted for input via stdin, and it's likely that users will munge
+   // the PATH within a terminal a way that makes our utilities unavailable
+   //
+   // TODO: could we just set an absolute path to the rpostback-askpass util?
+   std::string gitAskpass = core::system::getenv("GIT_ASKPASS");
+   if (gitAskpass == "rpostback-askpass")
+      core::system::unsetenv(&shellEnv, "GIT_ASKPASS");
+
+   std::string sshAskpass = core::system::getenv("SSH_ASKPASS");
+   if (sshAskpass == "rpostback-askpass")
+      core::system::unsetenv(&shellEnv, "SSH_ASKPASS");
+   
```

Comment:
I think something about this is behaving differently on Unix versus Windows, maybe because on Windows the handler is considered `rpostback-askpass.exe` rather than `rpostback-askpass`. And I notice that when I use the terminal in this local build, on Mac I do actually get prompted for a username/password in stdin, while on Windows I don't get prompted for username/password. Also, on Mac, I get the value of `rpostback-askpass` when I do `Sys.getenv("GIT_ASKPASS")` or `Sys.getenv("SSH_ASKPASS")` while on Windows, these two environment variables appear to be unset, not just in the terminal but directly in the R console with `Sys.getenv`.  So I'm a little confused about the differences in behavior here

Reply:
> I think something about this is behaving differently on Unix versus Windows, maybe because on Windows the handler is considered rpostback-askpass.exe rather than rpostback-askpass.

Can you clarify? The various helper scripts are just regular bash scripts (without extensions) so it should be correct to use these names as-is here. Only the `rpostback` executable itself will be called `rpostback.exe` on Windows.

> And I notice that when I use the terminal in this local build, on Mac I do actually get prompted for a username/password in stdin, while on Windows I don't get prompted for username/password.

Does authentication succeed? If it does, that would probably imply the credential manager is using existing credentials; if not, what error are you seeing?

> Also, on Mac, I get the value of rpostback-askpass when I do Sys.getenv("GIT_ASKPASS") or Sys.getenv("SSH_ASKPASS") while on Windows, these two environment variables appear to be unset, not just in the terminal but directly in the R console with Sys.getenv

I think I know what the issue here is; I'm going to push a change.